Automattic/wp-calypso
# Framework: Upgrade to eslint-config-wpcalypso v0.6.0

This pull request seeks to upgrade our `eslint-config-wpcalypso` dependency to v0.6.0. Copying changelog notes below for ease of reading:

>- General: Update to `eslint-plugin-wpcalypso@3.0.1` ([see changelog](https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/CHANGELOG.md#v301-2016-10-21))
>- Added: [`wpcalypso/i18n-no-collapsible-whitespace`](https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/docs/rules/i18n-no-collapsible-whitespace.md) as error
>- Added: [`wpcalypso/i18n-no-this-translate`](https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/docs/rules/i18n-no-this-translate.md) as error
>- Added: [`wpcalypso/import-docblock`](https://github.com/Automattic/eslint-plugin-wpcalypso/blob/master/docs/rules/import-docblock.md) as error
>- Added: [`react/prefer-es6-class`](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/prefer-es6-class.md) as error
>- Added: [`func-call-spacing`](http://eslint.org/docs/rules/func-call-spacing) as error ("never")

All rules added above have been downgraded to warnings in Calypso due to preexisting issues.

__Testing Instructions:__

Verify that there are no lint issues or errors when running `npm run lint`

cc @deBhal @umurkontaci
